File: frameworks/rust/axum/Implementation/serial.rs
use std::sync::atomic::{AtomicU64, Ordering};

use axum::http::{HeaderName, HeaderValue};

const SERIAL: HeaderName = HeaderName::from_static("x-rb-serial");

static LAST: AtomicU64 = AtomicU64::new(0);

/// x-rb-serial: one counter for the whole process. A handler that writes it takes the next value,
/// so an answer a cache replays carries the value it was stored with.
pub fn fresh() -> [(HeaderName, HeaderValue); 1] {
    [(SERIAL, HeaderValue::from(LAST.fetch_add(1, Ordering::Relaxed) + 1))]
}

File: frameworks/java/helidon-se/Implementation/src/main/java/implementation/Serial.java
package implementation;

import java.util.concurrent.atomic.AtomicLong;

import io.helidon.http.HeaderName;
import io.helidon.http.HeaderNames;
import io.helidon.webserver.http.ServerResponse;

/**
 * x-rb-serial: one counter for the whole process. A handler that writes it increments it and
 * writes the new value, so an answer the cache replays carries the value it was stored with.
 */
public final class Serial {

    public static final HeaderName HEADER = HeaderNames.create("x-rb-serial");

    private static final AtomicLong LAST = new AtomicLong();

    private Serial() {}

    public static String next() {
        return Long.toString(LAST.incrementAndGet());
    }

    public static void write(ServerResponse response) {
        response.header(HEADER, next());
    }
}
